token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,149 @@
+"""
+A module discovers and invoke Token Dispenser Service (TDS) lambda
+"""
+import logging
+import json
+from typing import List
+from botocore.exceptions import ClientError
+import boto3
+lambda_client = boto3.client('lambda')
+ssm = boto3.client('ssm')
+# ssm parameter store default path for TDS ARN
+_DEFAULT_SSM_PATH_: str = '/service/token-dispenser'
+logger = logging.getLogger(__name__)
+
+
+def get_parameter_by_name(name: str) -> str:
+    """
+
+    Args:
+        name: the ssm name point to TDS arn
+
+    Returns: tds arn as string
+    """
+    response = ssm.get_parameter(
+        Name=name
+    )
+    logger.debug('get_parameter_by_name with name: %s, response: %s', name, response)
+    if 'Parameter' not in response:
+        raise ValueError(f"Error: Parameter not found for name: {name}")
+    return response['Parameter']['Value']
+
+
+def invoke_lambda(input_params_json, lambda_arn):
+    """
+    Invokes an AWS Lambda function with input parameters from a JSON string.
+
+    Args:
+        input_params_json (str): A JSON string containing input parameters.
+        lambda_arn (str): The ARN of the Lambda function to invoke.
+
+    Returns:
+        str: The JSON-formatted response from the Lambda function, or an error message.
+    """
+    try:
+        response = lambda_client.invoke(
+            FunctionName=lambda_arn,
+            InvocationType='RequestResponse',  # Synchronous invocation
+            Payload=input_params_json
+        )
+
+        payload = response['Payload'].read().decode('utf-8')
+
+        # Check for function error
+        if 'FunctionError' in response:
+            raise RuntimeError(f"Error: Lambda function error: {payload}")
+
+        return payload
+    except ClientError as client_err:
+        logger.exception("Lambda invoking error %s", client_err)
+        raise client_err
+    except Exception as general_err:  # pylint: disable=broad-exception-caught
+        logger.exception("An unexpected general error occurred "
+                         "while invoking TDS: %s", general_err)
+        raise general_err
+
+
+def validate_input(client_id: str, minimum_alive_secs: int) -> List[str]:
+    """ validate the user inputs """
+    err_msgs = []
+    if client_id is None or client_id.strip() == '' or (not isinstance(client_id, str)):
+        err_msgs.append('Error: client_id is required as a string')
+
+    if minimum_alive_secs is not None and (not isinstance(minimum_alive_secs, int)):
+        err_msgs.append('Minimum alive interval, if provided, must be an integer')
+    return err_msgs
+
+
+def get_tds_arn(ssm_name: str) -> str:
+    """
+    :param ssm_name:
+    :return:
+    arn:str     : ARN of TDS lambda
+    """
+    response = None  # Initialize response to None
```

Comment:
Don't need this function level since it's only used within the `if` block. Can remove